src/testing/mock.rs

```diff
@@ -182,26 +182,24 @@ impl Future for HonestHub {
 }
 
 #[derive(Clone)]
-pub(crate) struct Spawner {
-    handles: Arc<Mutex<Vec<tokio::task::JoinHandle<()>>>>,
-}
+pub(crate) struct Spawner {}
 
 impl SpawnHandle for Spawner {
     fn spawn(&self, _name: &str, task: impl Future<Output = ()> + Send + 'static) {
-        self.handles.lock().push(tokio::spawn(task))
+        tokio::spawn(task);
     }
-}
 
-impl Spawner {
-    pub(crate) async fn wait(&self) {
-        for h in self.handles.lock().iter_mut() {
-            let _ = h.await;
-        }
-    }
-    pub(crate) fn new() -> Self {
```

Comment:
I would suggest leaving the `new` even if it is trivial, future changes are likely to be simpler (just look at the files in which you had to change only this).

Reply:
Done.